### PR TITLE
Always --pull when building Dockerfile

### PR DESCRIPTION
**Pull request by sebrandon1**
Related to: https://github.com/test-network-function/cnf-certification-test/pull/1461

### CODE REVIEW

File: .github/workflows/cnf-test-partner-image.yaml [hunk 1]
```diff
@@ -40,7 +40,7 @@ jobs:
           swap-storage: true
 
       - name: 'Build the `cnf-test-partner` image'
-        run: docker build  -t $IMAGE_NAME:$IMAGE_TAG -t $IMAGE_NAME:$SPECIFIC_IMAGE_TAG . --no-cache -f $IMAGE_CONTAINER_FILE_PATH
+        run: docker build --pull -t $IMAGE_NAME:$IMAGE_TAG -t $IMAGE_NAME:$SPECIFIC_IMAGE_TAG . --no-cache -f $IMAGE_CONTAINER_FILE_PATH
```
Comment (jmontesi) [inline .github/workflows/cnf-test-partner-image.yaml:43]: The images used in the `Dockerfile` are versioned, do we need this pull?

File: .github/workflows/debug-partner-image.yaml [hunk 2]
```diff
@@ -41,7 +41,7 @@ jobs:
           swap-storage: true
 
       - name: 'Build the `debug-partner` image'
-        run: docker build --no-cache -f Dockerfile.debug-partner -t $IMAGE_NAME:$IMAGE_TAG .
+        run: docker build --pull --no-cache -f Dockerfile.debug-partner -t $IMAGE_NAME:$IMAGE_TAG .
```
Comment (jmontesi) [inline .github/workflows/debug-partner-image.yaml:44]: Same here.

**Review by rdavid — APPROVED** [review]

File: .github/workflows/cnf-test-partner-image.yaml (hunk 1 above)
Comment (sebrandon1) [reply .github/workflows/cnf-test-partner-image.yaml:43]: Good point, this is only really good for `latest`.  I will close.

**Comment by sebrandon1** [comment]
Closing as this is only really useful for `latest` images in the Dockerfiles.

**Pull request closed without merging**